packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -214,37 +218,37 @@ export function useGridDimensions(
     return Math.min(maximumPageSizeWithoutScrollBar, currentPage.rows.length);
   }, [apiRef, props.pagination, props.paginationMode, props.getRowHeight, rowHeight]);
 
-  const computeSizeAndPublishResizeEvent = React.useCallback(() => {
-    const rootEl = apiRef.current.rootElementRef?.current;
-    const mainEl = rootEl?.querySelector<HTMLDivElement>(`.${gridClasses.main}`);
```

Comment:
I guess the main problem here is that the main element ref is assigned first, but we don't have direct access to it - therefore we use a query selector.
Instead of passing the element to this method, would it make more sense to make sure that the main element is accessible here directly?
I've opened https://github.com/mui/mui-x/pull/8968 with a unit test